python/paddle/optimizer/optimizer.py

```diff
@@ -772,7 +772,7 @@ def _append_optimize_op(self, block, param_and_grad):
     def _create_param_lr(self, param_and_grad):
         # create learning rate tensor for every parameter
         param = param_and_grad[0]
-        if hasattr(param, 'optimize_attr'):
+        if hasattr(param, 'optimize_attr') and param.optimize_attr is not None:
```

Comment:
why change this?

Reply:
In old IR mode, param is defined in python end，we can use ```hasattr(param, 'optimize_attr')``` to judge whether has optimize_attr. but in pir mode, param is defined in C++ end, the "optimize_attr" is a interface defined by pybind, it always existed, we can only  judge whether has optimize_attr by ```param.optimize_attr is not None```.